### PR TITLE
Documentation modifications

### PR DESCRIPTION
**Pull request by esdrasjnr**
- modify README to install Sphinx by pip3
- add knot cloud hyperlink to quick-start instructions
- fix data_register and data_config function names at sample code
- fix dongle reset pin number

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,11 +2,16 @@ KNoT Documentation
 ==================
 
 ### How to build:
+- Make sure that pip3 is installed in your machine:
+	```bash
+	$ sudo apt-get install pip3
```
Comment (alla3) [inline README.md]: This command refers only to Debian based versions and I think the name of this package should be `python3-pip`.

File: README.md [hunk 2]
```diff
@@ -16,6 +16,7 @@ KNoT Documentation
 
 - Build KNoT Doc:
 	```bash
+	$ source ~/.profile
```
[the remark this answers is not in the document]
Comment (esdrasjnr) [reply README.md]: @alla3 done

File: doc/thing/thing-create-app.rst [hunk 3]
```diff
@@ -121,9 +121,9 @@ On ``loop()`` function you can include your application logic.
 
     ``loop()`` function must NOT be blocking.
 
-On ``setup()`` you can register your sensors/actuators using `knot_proxy_register() <thing-api.html#knot-proxy-register>`_ and configures which events should send data to cloud with `knot_proxy_set_config() <thing-api.html#knot-proxy-set-config>`_.
+On ``setup()`` you can register your sensors/actuators using `knot_data_register() <thing-api.html#knot-data-register>`_ and configures which events should send data to cloud with `knot_data_config() <thing-api.html#knot-data-config>`_.
 
-For each registered sensor/actuator you may want to create a ``changed_cb`` or ``pool_cb`` function, this function are passed as callback on `register <thing-api.html#knot-proxy-register>`_ function.
+For each registered sensor/actuator you may want to create a ``changed_cb`` or ``pool_cb`` function, this function are passed as callback on `register <thing-api.html#knot-data-register>`_ function.
```
[the remark this answers is not in the document]
Comment (esdrasjnr) [reply doc/thing/thing-create-app.rst]: @ramonhpr done!

File: doc/thing/thing-create-app.rst [hunk 4]
```diff
@@ -161,25 +164,24 @@ In this example a LED Sensor is registered with:
...
+   - variable_size = sizeof(led);
+   - write_cb = write_led;
+   - read_cb = NULL;
 
 And it is configured to send message to Cloud every time the value changes.
 
- - The ``changed_cb`` function gets a value for a `knot_proxy <thing-api.html#struct-knot-proxy>`_ and it can set on a sensor/actuator.
- - The ``pool_cb`` function sets information of sensor/actuator on a `knot_proxy <thing-api.html#struct-knot-proxy>`_.
+ - ``write_cb``: This function is used to write in the actuator the value of
+   the associated variable. It is executed after variable value is updated.
```
Comment (ramonhpr) [inline doc/thing/thing-create-app.rst]:
```suggestion
   the associated variable. It is executed after the variable value is updated remotely.
```

File: doc/thing/thing-create-app.rst [hunk 5]
```diff
@@ -161,25 +164,24 @@ In this example a LED Sensor is registered with:
...
+ - ``read_cb``: This function is used to write in variable the value readed
```
Comment (ramonhpr) [inline doc/thing/thing-create-app.rst]:
```suggestion
 - ``read_cb``: This function is used to write the value read from the sensor into the variable
```

File: doc/thing/thing-create-app.rst [hunk 6]
```diff
@@ -130,9 +130,9 @@ On ``loop()`` function you can include your application logic.
 
     ``loop()`` function must NOT be blocking.
 
-On ``setup()`` you can register your sensors/actuators using `knot_proxy_register() <thing-api.html#knot-proxy-register>`_ and configures which events should send data to cloud with `knot_proxy_set_config() <thing-api.html#knot-proxy-set-config>`_.
+On ``setup()`` you can register your sensors/actuators using `knot_data_register() <thing-api.html#knot-data-register>`_ and configures which events should send data to cloud with `knot_data_config() <thing-api.html#knot-data-config>`_.
 
-For each registered sensor/actuator you may want to create a ``changed_cb`` or ``pool_cb`` function, this function are passed as callback on `register <thing-api.html#knot-proxy-register>`_ function.
+For each registered sensor/actuator you may want to create a ``write_cb`` or ``read_cb`` function, this function are passed as callback on `register <thing-api.html#knot-data-register>`_ function.
```
Comment (lcbm) [inline doc/thing/thing-create-app.rst]: Same as before! The changes you made seem good :smile: 

Though, there are a few grammar mistakes that you may want to consider fixing: 

1. `function, this function... `: `this` is a **singular** proximal demonstrative pronoun. I believe the phrase means to refer to both `write_cb` and `read_cb`; therefore, one of the possible corrections would be `function, these functions...`.
2. Also in `function, this functions...`: the comma shouldn't be there ((you can find some rules on comma usage [here](https://www.businessinsider.com/a-guide-to-proper-comma-use-2013-9)). A more suitable alternative would be `function. These functions are then passed as callbacks on...`. If you really must use a comma, then the phrasing should change to something like `function, which will be passed as callbacks on...`

File: doc/thing/thing-create-app.rst [hunk 7]
```diff
@@ -130,9 +130,9 @@ On ``loop()`` function you can include your application logic.
 
     ``loop()`` function must NOT be blocking.
 
-On ``setup()`` you can register your sensors/actuators using `knot_proxy_register() <thing-api.html#knot-proxy-register>`_ and configures which events should send data to cloud with `knot_proxy_set_config() <thing-api.html#knot-proxy-set-config>`_.
+On ``setup()`` you can register your sensors/actuators using `knot_data_register() <thing-api.html#knot-data-register>`_ and configures which events should send data to cloud with `knot_data_config() <thing-api.html#knot-data-config>`_.
```
[the remark this answers is not in the document]
Comment (esdrasjnr) [reply doc/thing/thing-create-app.rst]: Done!

File: doc/thing/thing-create-app.rst (hunk 6 above)
Comment (esdrasjnr) [reply doc/thing/thing-create-app.rst]: Done!

**Pull request merged**